cloudify_agent/installer/__init__.py

```diff
@@ -341,16 +342,21 @@ def create_custom_env_file_on_target(self, environment):
             return None
 
     def download(self, url, destination=None):
+        local_cert_file = self.cloudify_agent.get('local_rest_cert_file')
+        verify_cert = self.cloudify_agent.get('verify_rest_certificate')
+        if verify_cert is None:
+            raise exceptions.AgentInstallerConfigurationError(
+                'Missing verify_rest_certificate')
         if self.cloudify_agent['windows']:
-            return self.runner.download(url,
-                                        output_path=destination,
-                                        skip_verification=True)
+            return self.runner.download(
+                url,
+                output_path=destination,
+                skip_verification=not verify_cert)
         else:
-            local_cert_file = self.cloudify_agent.get('local_rest_cert_file')
             return self.runner.download(
                 url,
                 output_path=destination,
-                skip_verification=not bool(local_cert_file),
+                skip_verification=not verify_cert,
```

Comment:
yes, Idan asked me to accept lukasz's comment, which makes it less relying on the creation of the cert file and clearer